studio/LonaStudio/Workspace/FileNavigator.swift

```diff
@@ -185,6 +185,36 @@ class FileNavigator: NSBox {
                 menu.addItem(NSMenuItem.separator())
             }
 
+            if NSURL(fileURLWithPath: path).pathExtension == "component"{
+                menu.addItem(NSMenuItem(title: "Duplicate As...", onClick: {
+                    var saveURL: String
+                    let dialog = NSSavePanel()
```

Comment:
I just pulled down this branch and tried it out. I think the most common use case for this feature is to duplicate a component into the same directory. Let's set `dialog.directoryURL` to make that easier. We can do this with `dialog.directoryURL = URL(fileURLWithPath: path).deletingLastPathComponent()`. I didn't think of this before, but I think it'll help a lot.